restaurant.rb

```diff
@@ -0,0 +1,61 @@
+require 'date'
+require 'httparty'
+require 'nokogiri'
+require 'active_support/core_ext/string/multibyte'
+require 'active_support/multibyte/chars'
+
+class Restaurant
+
+  def menu_text
+    raise "No restaurant name specified" unless @name
+    menu_text = menu.map {|item| menu_item_text(item) }.join("\n")
+
+    "*#{title_text}*\n#{menu_text}"
+  end
+
+  def command_name
+    self.class.name.downcase
+  end
+
+  def latest_workday
+    # by default, restaurants work monday (1) to friday (5)
+    [Date.today.wday, 5].min
+  end
+
+  def latest_workdate
+    (Date.today - Date.today.wday) + latest_workday
+  end
+
+  def self.descendants
+    ObjectSpace.each_object(Class).select {|klass| klass < self }
+  end
+
+protected
+
+  def page_body
+    raise "No url specified" unless @page_url
+    response = HTTParty.get(@page_url)
+    Nokogiri::HTML(response.body)
+  end
+
+  def title_text
+    date = latest_workdate.strftime("%A, %d. %m. %Y")
```

Comment:
It returns the last workdate. Because there are no menus on a saturday, it
returns friday. The workday is limited to mon-fri by default, but you can
override it. Is it not working? Which date are you getting?
On 17 May 2015 20:48, "radiokills" <notifications@github.com> wrote:

> In restaurant.rb
> <https://github.com/radiokills/foodbot/pull/2#discussion_r30473305>:
>
> > +  end
> > +
> > +  def self.descendants
> > +    ObjectSpace.each_object(Class).select {|klass| klass < self }
> > +  end
> > +
> > +protected
> > +
> > +  def page_body
> > +    raise "No url specified" unless @page_url
> > +    response = HTTParty.get(@page_url)
> > +    Nokogiri::HTML(response.body)
> > +  end
> > +
> > +  def title_text
> > +    date = latest_workdate.strftime("%A, %d. %m. %Y")
>
> What about latest workday? If I'm trying this on Saturday I'll get wrong
> date here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/radiokills/foodbot/pull/2/files#r30473305>.
>